lib/mayaUsd/ufe/UsdUndoMaterialCommands.cpp

```diff
@@ -301,138 +301,158 @@ std::string UsdUndoAssignNewMaterialCommand::resolvedMaterialScopeName()
+            scopeItem, shaderNodeDef->GetFamily().GetString(), "Material");
+        if (!createMaterialCmd) {
+            markAsFailed();
+            return;
+        }
+        createMaterialCmd->execute();
+        _createMaterialCmdIdx = _cmds->cmdsList().size();
+        _cmds->append(createMaterialCmd);
+        if (!createMaterialCmd->newPrim()) {
+            // The _createMaterialCmd will have emitted errors.
+            markAsFailed();
+            return;
+        }
 
-                //
-                // 3. Create the Shader if it does not exist:
-                //
-                materialItem = std::dynamic_pointer_cast<UsdSceneItem>(
-                    Ufe::Hierarchy::createItem(createMaterialCmd->newUfePath()));
-                auto createShaderCmd = UsdUndoCreateFromNodeDefCommand::create(
-                    shaderNodeDef, materialItem, shaderNodeDef->GetFamily().GetString());
-                if (!createShaderCmd) {
-                    markAsFailed();
-                    return;
-                }
-                createShaderCmd->execute();
-                _cmds->append(createShaderCmd);
-                if (!createShaderCmd->insertedChild()) {
-                    // The _createShaderCmd will have emitted errors.
-                    markAsFailed();
-                    return;
-                }
+        //
+        // 3. Create the Shader:
+        //
+        UsdSceneItem::Ptr materialItem = std::dynamic_pointer_cast<UsdSceneItem>(
+            Ufe::Hierarchy::createItem(createMaterialCmd->newUfePath()));
+        auto createShaderCmd = UsdUndoCreateFromNodeDefCommand::create(
+            shaderNodeDef, materialItem, shaderNodeDef->GetFamily().GetString());
+        if (!createShaderCmd) {
+            markAsFailed();
+            return;
+        }
+        createShaderCmd->execute();
+        _cmds->append(createShaderCmd);
+        if (!createShaderCmd->insertedChild()) {
+            // The _createShaderCmd will have emitted errors.
+            markAsFailed();
+            return;
+        }
 
-                //
-                // 4. Connect the Shader to the material:
-                //
-                if (!connectShaderToMaterial(
-                        createShaderCmd->insertedChild(), createMaterialCmd->newPrim(), _nodeId)) {
-                    markAsFailed();
-                    return;
-                }
-            }
+        //
+        // 4. Connect the Shader to the material:
+        //
+        if (!connectShaderToMaterial(
+                createShaderCmd->insertedChild(), createMaterialCmd->newPrim(), _nodeId)) {
+            markAsFailed();
+            return;
+        }
 
-            //
-            // 5. Bind the material to the parent primitive:
-            //
+        //
+        // 5. Bind the material to all selected primitives in the stage:
+        //
+        for (const auto& parentPath : selectedPaths) {
```

Comment:
I moved steps 1 through 4 out of the inner loop. These steps were only executed in the first iteration of the inner loop anyway, so moving them out reduces unnecessary indentation and should improve readability.